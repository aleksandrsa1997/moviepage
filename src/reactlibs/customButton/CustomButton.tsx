import React from 'react';
import Button from '@material-ui/core/Button';
import { useStyles } from './styles';
import { useTranslation } from 'react-i18next';

export type TProps = {
    size?: 	'large'| 'medium'| 'small'
    color?: 'default' | 'inherit' | 'primary' | 'secondary';
    variant?:  'contained' | 'outlined' | 'text';
    onClick?: () => void;
    textKey?: string;
    disabled?: boolean;
};

const CustomButton = ({
    size= 'small',
    color = 'inherit',
    variant = 'contained',
    textKey = 'button',
    onClick
}: TProps) => {

    const { t } = useTranslation();
    const classes = useStyles();

    return (
        <Button
            size={size}
            color={color}
            variant={variant}
            onClick={onClick}
            className={classes.text}
        >
            {t(textKey)}
        </Button>
    );
}

export default CustomButton;