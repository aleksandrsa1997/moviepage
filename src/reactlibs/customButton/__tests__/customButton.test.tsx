import React from 'react';
import { act } from 'react-dom/test-utils';
import CustomButton, { TProps } from '../CustomButton';
import {mount, ReactWrapper} from 'enzyme';

jest.mock('react-i18next', () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

describe('CustomButton tests', () => {
    let props: TProps;
    let component: ReactWrapper<any, React.Component["state"], React.Component>;

    beforeEach(()=> {
        props = {
            color: 'inherit',
            variant: 'contained',
            textKey: 'button',
            onClick: jest.fn()
        }
        component = mount(<CustomButton {...props}/>);
    })

    it('should render snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should call onClick function', () => {
        act(() => {
            component.find('CustomButton').at(0).getElement().props.onClick();
        })
        component.update();
        expect(props.onClick).toHaveBeenCalled();
    })
});