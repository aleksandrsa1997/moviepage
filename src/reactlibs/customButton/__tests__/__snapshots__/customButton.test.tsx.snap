// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton tests should render snapshot 1`] = `ReactWrapper {}`;
